### ISSUE TITLE
nerdctl rmi can be broken by using tag names with repo short sha

### ISSUE THREAD

**Issue by apostasie** [containerd/nerdctl#3016]
### Description

When an image uses a tag that is a short version of another (unrelated) image repo sha, nerdctl rmi will break down.

This is another aspect of one of the many issues with our current image lookup implementation.

### Steps to reproduce the issue

```
nerdctl pull busybox
nerdctl pull alpine
shortid="$(nerdctl inspect busybox | jq -rc .[0].RepoDigests[0])"
shortid="${shortid##*:}"
shortid="${shortid:0:8}"
nerdctl tag alpine "$shortid"
nerdctl rmi "$shortid"
```

### Describe the results you received and expected

```
FATA[0000] 1 errors:
multiple IDs found with provided prefix: 5eef5ed3
```

Should have just worked and removed the tagged version of alpine.

### What version of nerdctl are you using?

1.7.6

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

**Comment by apostasie** [comment]
Can we reopen?
#3017 was not meant to fix this (yet).

**Comment by apostasie** [comment]
@yankay @AkihiroSuda can we reopen this?

**Comment by haytok** [comment]
Hi, team

I want to work on this issue, so could you assign this ?

**Comment by apostasie** [comment]
Fixed by #3519
